File: 0206-reverse-linked-list/0206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode current = head;
        ListNode reverse = null; 
        while (current!= null){
            ListNode newNode = current.next;
        current.next=reverse;
        reverse = current;
        current = newNode;
        System.out.print(reverse.val);

        }
        // System.out.print(head.next); 
    return reverse;
    }
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        #Solution 1 
        # prev = None
        # curr = head
        # while curr:
        #     temp = curr.next
        #     curr.next = prev
        #     prev = curr
        #     curr = temp
            
        # return prev
        #SOLUTION 2 USING RECURSION
        if not head or not head.next:
            return head
        current = self.reverseList(head.next)
        head.next.next = head
        head.next = None
        return current